examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,272 @@
+=========================================================
+Ridge coefficients as a function of the L2 Regularization
+=========================================================
 
+A model that overfits learns the training data too well, capturing both the
+underlying patterns and the noise in the data. However, when applied to unseen
+data, the learned associations may not hold. We normally detect this when we
+apply our trained predictions to the test data and see the performance drop
+significantly compared to the training data.
+
+One way to overcome overfitting is through regularization. Regularization
+penalizes large training weights, forcing the model to shrink certain
+coefficients. Regularization reduces a model's reliance on specific information
+obtained from the training samples.
+
+This example illustrates how L2 regularization in a
+:class:`~sklearn.linear_model.Ridge` regression affects a model's performance by
+adding a penalty term to the Loss that increases with the coefficients (β).
+
+The regularized loss function is given by: :math:`\\mathcal{L}(X, y, \\beta) = \\|
+y - X \\beta \\|^{2}_{2} + \\alpha \\| \\beta \\|^{2}_{2}`
+
+where :math:`X` is the input data, :math:`y` is the target variable,
+:math:`\\beta` is the vector of coefficients associated with the features, and
+:math:`\\alpha` is the regularization strength.
+
+The regularized loss function aims to balance the trade-off between accurately
+predicting the training set and to prevent overfitting.
+
+In this regularized loss, the left-hand side (:math:`\\|y - X\\beta\\|^{2}_{2}`)
+measures the squared difference between the actual target variable, :math:`y`,
+and the predicted values. Minimizing this term alone could lead to overfitting,
+as the model may become too complex and sensitive to noise in the training data.
+
+To address overfitting, Ridge regularization adds a constraint, called a penalty
+term, (:math:`\\alpha \\| \\beta\\|^{2}_{2}`) to the loss function. This penalty
+term is the sum of the squares of the model's coefficients, multiplied by the
+regularization strength :math:`\\alpha`. By introducing this constraint, Ridge
+regularization discourages any single coefficient :math:`\\beta_{i}` from taking
+an excessively large value and encourages smaller and more evenly distributed
+coefficients. Higher values of :math:`\\alpha` force the coefficients towards
+zero. However, an excessively high :math:`\\alpha` can result in an underfit
+model that fails to capture important patterns in the data.
+
+Therefore, the regularized loss function combines the prediction accuracy term
+and the penalty term. By adjusting the regularization strength, practitioners
+can fine-tune the degree of constraint imposed on the weights, training a model
+capable of generalizing well to unseen data while avoiding overfitting.
+
+For a comparison between Ridge and Huber Regularization see
+:doc:`~examples.linear_model.plot_huber_vs_ridge`
```

Comment:
To reference properly the example, we need the following.

```suggestion
For a comparison between Ridge and Huber Regularization see
:ref:`sphx_glr_auto_examples_linear_model_plot_huber_vs_ridge.py`
```

Reply:
I think that I would need refer to the example here but in the conlusion. I will put a comment there with a proposal.